aloscene/scene_flow.py

```diff
@@ -0,0 +1,187 @@
+from aloception.aloscene import Depth, CameraIntrinsic, Mask, Flow
+import aloception.aloscene as aloscene
+from typing import Union
+import numpy as np
+import torch
+import torch.nn.functional as F
+
+
+def load_scene_flow(path: str) -> torch.Tensor:
+    with open(path, "rb") as file:
+        data = np.load(file)
+        return torch.from_numpy(data)
+
+
+class SceneFlow(aloscene.tensors.SpatialAugmentedTensor):
+    """
+    Scene flow map
+
+    Parameters
+    ----------
+    x : str
+        load scene flow from a numpy file
+    """
+
+    @staticmethod
+    def __new__(cls, x, occlusion: Union[Mask, None], *args, names=("C", "H", "W"), **kwargs):
+        if isinstance(x, str):
+            # load flow from path
+            x = load_scene_flow(x)
+            names = ("C", "H", "W")
+
+        tensor = super().__new__(cls, x, *args, names=names, **kwargs)
+        tensor.add_child("occlusion", occlusion, align_dim=["B", "T"], mergeable=True)
+        return tensor
+
+    def __init__(self, x, *args, **kwargs):
+        super().__init__(x)
+
+    @classmethod
+    def from_optical_flow(
+        cls,
+        optical_flow: Flow,
+        depth: Depth,
+        next_depth: Depth,
+        intrinsic: CameraIntrinsic,
+        sampling: str = "bilinear",
+    ):
+        """Create scene flow from optical flow, depth a T, depth at T + 1 and the intrinsic
+
+        Parameters
+        ----------
+        optical flow: aloscene.Flow
+            The optical flow at T.
+        depth: aloscene.Depth
+            The depth at T.
+        next_depth: aloscene.Depth
+            The depth at T + 1
+        intrinsic : aloscene.CameraIntrinsic
+            The intrinsic of the image at T.
+        sampling: str
+            The sampling method to use for the scene flow.
+        """
+        has_batch = True if len(optical_flow.names) == 4 else False
```

Comment:
names can be (T, C, H, W). The correct way to check is : `has_batch = "B" in names`.
However, with the two assertions below it's correct.